### ISSUE TITLE
Restore ApolloServerStandalone functionality

### ISSUE THREAD

**Issue by glasser** [apollographql/apollo-server#6085]
The batteries-included `apollo-server` package has been deleted from the `version-4` branch. Its functionality should be restored as a new class with a different name in the main package, like `ApolloServerStandalone` or something.

**Comment by glasser** [comment]
Some of the cors tests deleted in https://github.com/apollographql/apollo-server/commit/81c7f7b2692d57bb45e3755dc44e7e3bfe1a03c4 will probably be useful for the standalone server.

**Comment by glasser** [comment]
@trevor-scheer and I ran into some issues related to generic types and TContext trying to make this a class. My current idea is to make this a function instead (pretty similar to what @abernix originally proposed in #3184 though with a bit more async-ness). This noun-named `httpServer` is analogous to `expressMiddleware`.

```
// or maybe you have to deep-import httpServer to avoid importing `express` otherwise?
// or maybe the `express` import is a runtime import() instead.
import { ApolloServer, httpServer } from '@apollo/server';
const server = new ApolloServer<MyContext>({ typeDefs, resolvers });
await httpServer(server, {
  context: async ({ req, res }) => new MyContext(req),
}).listen(1234);
```

where `httpServer` is a function that synchronously returns something like

```
class ApolloHttpServer {
  async listen(...args): Promise<{url: string}> {
  }
}
```

**Comment by glasser** [comment]
@trevor-scheer done?

**Comment by trevor-scheer** [comment]
Closed via #6420
